### ISSUE TITLE
False positive `TRY300` when there is no `except` clause

### ISSUE THREAD

**Issue by Czaki** [charliermarsh/ruff#3632]
In napari we have such a function https://github.com/napari/napari/blob/ab885957fb5a2220a3078851ad7dc97a7a82cc05/napari/utils/misc.py#L253

```python
def formatdoc(obj):
    """Substitute globals and locals into an object's docstring."""
    frame = inspect.currentframe().f_back
    try:
        obj.__doc__ = obj.__doc__.format(
            **{**frame.f_globals, **frame.f_locals}
        )
        return obj
    finally:
        del frame
```

The `TRY300` is reported on `return obj`. But the else clause is available only when the `except` clause is present (there needs to be at least one). So `TRY300` should not be reported if there is no except clause. Or it should have different text describing that. 

Currently:

```
TRY300 Consider moving this statement to an `else` block
```

Possible:

```
TRY300 Consider moving this statement after `try: finally:`
```

**Comment by JonathanPlasse** [comment]
I would like to work on it.
